common/scala/src/main/scala/whisk/core/database/DocumentFactory.scala

```diff
@@ -149,15 +108,12 @@ trait DocumentFactory[W] extends MultipleReadersSingleWriterCache[W, DocInfo] {
       implicit val ec = db.executionContext
 
       val key = CacheKey(doc)
+      val docInfo = doc.docinfo
 
-      cacheUpdate(doc, key, db.put(doc) map { docinfo =>
-        doc match {
-          // if doc has a revision id, update it with new version
-          case w: DocumentRevisionProvider => w.revision[W](docinfo.rev)
-        }
-        docinfo
+      cacheUpdate(doc, key, db.put(doc) map { newDocInfo =>
+        doc.revision[W](newDocInfo.rev)
```

Comment:
I know I used the copy but thinking if instead we should have a .revision on docinfo, or take the new doc info from the revised doc on line 114 to avoid an unnecessary copy.